src/content-tags-drawer/ContentTagsDrawer.jsx

```diff
@@ -26,6 +32,42 @@ const ContentTagsDrawer = () => {
 
   const org = extractOrgFromContentId(contentId);
 
+  const [stagedContentTags, setStagedContentTags] = useState({});
+
+  // Add a content tags to the staged tags for a taxonomy
+  const addStagedContentTag = useCallback((taxonomyId, addedTag) => {
+    setStagedContentTags(prevStagedContentTags => {
+      const updatedStagedContentTags = cloneDeep(prevStagedContentTags);
+      if (!updatedStagedContentTags[taxonomyId]) {
+        updatedStagedContentTags[taxonomyId] = [];
+      }
+      updatedStagedContentTags[taxonomyId].push(addedTag);
+      return updatedStagedContentTags;
+    });
+  }, [setStagedContentTags]);
+
+  // Remove a content tag from the staged tags for a taxonomy
+  const removeStagedContentTag = useCallback((taxonomyId, tagLabel) => {
+    setStagedContentTags(prevStagedContentTags => {
+      const updatedStagedContentTags = cloneDeep(prevStagedContentTags);
+      let stagedTaxonomyTags = updatedStagedContentTags[taxonomyId];
+      if (stagedTaxonomyTags && stagedTaxonomyTags.length) {
+        stagedTaxonomyTags = stagedTaxonomyTags.filter((t) => t.label !== tagLabel);
+        updatedStagedContentTags[taxonomyId] = stagedTaxonomyTags;
+      }
+      return updatedStagedContentTags;
+    });
```

Comment:
```suggestion
    setStagedContentTags(prevStagedContentTags => ({
      ...prevStagedContentTags,
      [taxonomyId]: prevStagedContentTags[taxonomyId].filter((t) => t.label !== tagLabel),
    }));
```

My personal rule of thumb: if you see `lodash` or `underscore`, there's probably a better way to do it using newer, vanilla JS without a dependency.

Note that this doesn't modify any existing objects - it strictly creates new ones (a new object and a new array) or re-uses existing ones unchanged (the arrays for any taxonomies that aren't modified).

Also: is `label` guaranteed to be unique here? It's the same as `value`, essentially?

Reply:
Makes sense, removed the cloneDeep and updated to match the standard style. I've also switched it from using `label` to using the `value` since that has a stronger guarantee of being unique.